src/webp/vp8.rs

```diff
@@ -1665,8 +1665,7 @@ fn avg3(left: u8, this: u8, right: u8) -> u8 {
 }
 
 fn avg2(this: u8, right: u8) -> u8 {
-    let avg = (u16::from(this) + u16::from(right) + 1) >> 1;
-    avg as u8
+    (this>>1) + (right>>1) + ((this | right) & 1)
```

Comment:
Do you also have some data showing that this is actually an improvement?

Reply:
And please don't simply benchmark this function alone, this is sure to get inline and hopefully optimized heavily at the inline site. Instead, benchmark some function that makes use of `avg2`.

Reply:
Edit: better yet, benchmark `predict_4x4` directly under different modes. It would probably be difficult (and mask the influence of the change) to move the benchmark up to `intra_predict` so that might be the best target for a benchmark.